app/uk/gov/hmrc/economiccrimelevyregistration/navigation/AmlSupervisorPageNavigator.scala

```diff
@@ -16,44 +16,54 @@
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
 
 import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
 
-class AmlSupervisorPageNavigator @Inject() () extends PageNavigator {
+class AmlSupervisorPageNavigator @Inject() (auditConnector: AuditConnector)(implicit ec: ExecutionContext)
+    extends AsyncPageNavigator {
 
   override protected def navigateInNormalMode(
     registration: Registration
-  ): Call =
+  )(implicit request: RequestHeader): Future[Call] =
     registration.amlSupervisor match {
       case Some(amlSupervisor) =>
         amlSupervisor.supervisorType match {
-          case t @ (GamblingCommission | FinancialConductAuthority) => registerWithGcOrFca(t)
-          case Hmrc | Other                                         => routes.RelevantAp12MonthsController.onPageLoad(NormalMode)
+          case t @ (GamblingCommission | FinancialConductAuthority) => registerWithGcOrFca(t, registration)
+          case Hmrc | Other                                         => Future.successful(routes.RelevantAp12MonthsController.onPageLoad(NormalMode))
         }
-      case _                   => routes.NotableErrorController.answersAreInvalid()
+      case _                   => Future.successful(routes.NotableErrorController.answersAreInvalid())
     }
 
   override protected def navigateInCheckMode(
     registration: Registration
-  ): Call =
+  )(implicit request: RequestHeader): Future[Call] =
     registration.amlSupervisor match {
       case Some(amlSupervisor) =>
         amlSupervisor.supervisorType match {
-          case t @ (GamblingCommission | FinancialConductAuthority) => registerWithGcOrFca(t)
-          case Hmrc | Other                                         => routes.CheckYourAnswersController.onPageLoad()
+          case t @ (GamblingCommission | FinancialConductAuthority) => registerWithGcOrFca(t, registration)
+          case Hmrc | Other                                         => Future.successful(routes.CheckYourAnswersController.onPageLoad())
         }
-      case _                   => routes.NotableErrorController.answersAreInvalid()
+      case _                   => Future.successful(routes.NotableErrorController.answersAreInvalid())
     }
 
-  private def registerWithGcOrFca(amlSupervisorType: AmlSupervisorType): Call =
+  private def registerWithGcOrFca(amlSupervisorType: AmlSupervisorType, registration: Registration): Future[Call] =
     amlSupervisorType match {
-      case GamblingCommission        => routes.RegisterWithGcController.onPageLoad()
-      case FinancialConductAuthority => routes.RegisterWithFcaController.onPageLoad()
-      case _                         => routes.NotableErrorController.answersAreInvalid()
+      case GamblingCommission        =>
+        auditConnector
+          .sendExtendedEvent(RegistrationNotLiableAuditEvent(registration).extendedDataEvent)
+          .map(_ => routes.RegisterWithGcController.onPageLoad())
```

Comment:
could perhaps put this into a private def to shorten it a bit as it's the same code used below? i.e. `sendNotLiableAuditEvent`